### ISSUE TITLE
[Frontend] Sign Up Page of Frontend project should be implemented

### ISSUE THREAD

**Issue by CenekSanzak** [bounswe/2021SpringGroup2#155]
In order to allow our users to sign up to our project, there should be a sign up form in the frontend website. It should get specified information from the user and communicate with the backend to complete the process. It should also do input validation.

**Comment by CenekSanzak** [comment]
[Feature #155](https://github.com/bounswe/2021SpringGroup2/tree/feature-%23155) branch is created for this feature request.

**Comment by CenekSanzak** [comment]
I created the signup page using mui components. I also added input validation on the client side using the [joi](https://www.npmjs.com/package/joi) package on the npm here is the screenshot of the signup page on my localhost. Implementation of this page is completed. It will use the backend endpoints when they are ready.

![image](https://user-images.githubusercontent.com/36790615/140988397-0a61c427-891f-493d-b5ac-979b2eccde25.png)

**Comment by CenekSanzak** [comment]
Pull Request for this feature #157 is created and waits for reviews.

**Comment by CenekSanzak** [comment]
Created a pull request for this issue, it is reviewed by some team members, it is updated after some reviews and it is merged with the frontend branch.

**Comment by CenekSanzak** [comment]
This issue is reponed for creating controllers to use backend endpoints

**Comment by CenekSanzak** [comment]
It will use the [API Documentation](https://github.com/bounswe/2021SpringGroup2/wiki/Rebound-API-documentation)

**Comment by CenekSanzak** [comment]
The API from the documentation is: 
## Register

`POST 'api/user/create/'`

* Author: Ege Can Kaya

* Definition: Allows a user to register

* Example Response Value:
    ```javascript
    {
        "email": "egekaya@somemail.com",
        "username": "egekaya"
    }
    ```
* Example Failure Value:
    ```javascript
    {
        "username": [
            "This field is required."
        ],
        "password": [
            "This field is required."
        ]
    }
    ```
* Parameters
    | Parameter | Description | Parameter Type | Data Type | 
    | :---: | :---: | :---: | :---: |
    | username | Username of registering user | Body | String |
    | email | Email address of registering user | Body | String |
    | password | Password of registering user | Body | String |


* Response Messages
  * 201: Created.
  * 400: Bad request.

**Comment by CenekSanzak** [comment]
It is done and merged in the pull request #193
